reaper/cluster.go

```diff
@@ -0,0 +1,243 @@
+package reaper
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"net/url"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type Cluster struct {
+	Name           string
+	JmxUsername    string
+	JmxPasswordSet bool
+	Seeds          []string
+	NodeState      NodeState
+}
+
+type NodeState struct {
+	GossipStates []GossipState
+}
+
+type GossipState struct {
+	SourceNode    string
+	EndpointNames []string
+	TotalLoad     float64
+	DataCenters   map[string]DataCenterState
+}
+
+type DataCenterState struct {
+	Name  string
+	Racks map[string]RackState
+}
+
+type RackState struct {
+	Name      string
+	Endpoints []EndpointState
+}
+
+type EndpointState struct {
+	Endpoint       string
+	DataCenter     string
+	Rack           string
+	HostId         string
+	Status         string
+	Severity       float64
+	ReleaseVersion string
+	Tokens         string
+	Load           float64
+}
+
+type GetClusterResult struct {
+	Cluster *Cluster
+	Error   error
+}
+
+type RepairSchedule struct {
+	Id                string    `json:"id"`
+	Owner             string    `json:"owner,omitempty"`
+	State             string    `json:"state,omitempty"`
+	Intensity         float64   `json:"intensity,omitempty"`
+	ClusterName       string    `json:"cluster_name,omitempty"`
+	KeyspaceName      string    `json:"keyspace_name,omitempty"`
+	RepairParallism   string    `json:"repair_parallelism,omitempty"`
+	IncrementalRepair bool      `json:"incremental_repair,omitempty"`
+	ThreadCount       int       `json:"repair_thread_count,omitempty"`
+	UnitId            string    `json:"repair_unit_id,omitempty"`
```

Comment:
Same question as above.

Reply:
Indeed... would `RepairUnitId` be more explicit?

Reply:
Will add the prefix.

Reply:
Added in f373d62.